drivers/pwm/pwm_nxp_flexio.c

```diff
@@ -101,6 +101,8 @@ static int pwm_nxp_flexio_set_cycles(const struct device *dev,
 	FLEXIO_Type *flexio_base = (FLEXIO_Type *)(config->flexio_base);
 	struct nxp_flexio_child *child = (struct nxp_flexio_child *)(config->child);
 	enum pwm_nxp_flexio_polarity polarity;
+	uint32_t duty_cycle = period_cycles - pulse_cycles;
```

Comment:
this variable name doesn't make sense, what you are calculating is inactive cycles, not the duty cycle, which is a percentage of active time of the period, so this is opposite the name for multiple reasons

Reply:
Sorry, I will rename it on the next push, it must have went through an iterative process.

Reply:
Removed, variable no longer needed.